cloud_chatops/lib/get_github_prs.py

```diff
@@ -0,0 +1,112 @@
+"""
+This module handles the HTTP requests and formatting to the GitHub REST Api.
+It will get all open pull requests in provided repositories.
+"""
+
+from typing import List, Dict
+import requests
+from lib.read_data import get_token
+from lib.pr_dataclass import PrData
+from lib.custom_exceptions import RepoNotFound, UnknownHTTPError, BadGitHubToken
+
+
+class GetGitHubPRs:
+    # pylint: disable=R0903
+    # Disabling this as in the future there is likely to be more public functions.
+    """
+    This class handles getting the open PRs from the GitHub Rest API.
+    """
+
+    def __init__(self, repos: List[str], owner: str):
+        """
+        This method initialises the class with the following attributes.
+        :param repos: A list of repositories to get pull requests for.
+        :param repos: The owner of the above repositories.
+        """
+        self.repos = repos
+        self.owner = owner
+        self._http_handler = HTTPHandler()
+
+    def run(self) -> List[PrData]:
+        """
+        This method is the entry point to the class.
+        It runs the HTTP request methods and returns the responses.
+        :return: The responses from the HTTP requests.
+        """
+        responses = self._request_all_repos_http()
+        return self._responses_to_dataclasses(responses)
+
+    def _request_all_repos_http(self) -> List[Dict]:
+        """
+        This method starts a request for each repository and returns a list of those PRs.
+        :return: A dictionary of repos and their PRs.
+        """
+        responses = []
+        for repo in self.repos:
+            url = f"https://api.github.com/repos/{self.owner}/{repo}/pulls"
+            responses += self._http_handler.make_request(url)
+        return responses
+
+    @staticmethod
+    def _responses_to_dataclasses(responses: List[Dict]) -> List[PrData]:
```

Comment:
missing docstring

Reply:
function could be named better - like parse_responses or build_responses

Reply:
👍